src/LogicItLab/Salesforce/MapperBundle/WsdlValidator.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace LogicItLab\Salesforce\MapperBundle;
+
+use LogicItLab\Salesforce\MapperBundle\Annotation\AnnotationReader;
+use LogicItLab\Salesforce\MapperBundle\Exception\FileHandlerNotInitializedException;
+use Symfony\Component\Finder\Finder;
+
+class WsdlValidator
+{
+    /** @var AnnotationReader */
+    private $annotationReader;
+
+    /** @var string */
+    private $srcPath;
+
+    /** @var string */
+    private $relativePath;
+
+    /** @var string */
+    private $wsdlPath;
+
+    /**
+     * @param AnnotationReader $annotationReader
+     * @param string $srcPath
+     * @param string $relativePath
+     * @param string $wsdlPath
+     * @codeCoverageIgnore
+     */
+    public function __construct(AnnotationReader $annotationReader, string $srcPath, string $relativePath, string $wsdlPath)
+    {
+        $this->annotationReader = $annotationReader;
+        $this->srcPath = $srcPath;
+        $this->relativePath = $relativePath;
+        $this->wsdlPath = $wsdlPath;
+    }
+
+    /**
+     * @return array
+     * @throws FileHandlerNotInitializedException
+     * @throws \ReflectionException
+     */
+    public function retrieveMissingFields(): array
+    {
+        $missingFields = [];
+
+        $fileHandler = fopen($this->wsdlPath, "r");
+        $this->initializeFileHandler($fileHandler);
+
+        foreach ($this->getAllClassAnnotations() as $annotation) {
```

Comment:
I'd suggest avoiding multi-level nesting for easier readability and maintainability.